lib/spack/spack/main.py

```diff
@@ -777,13 +778,15 @@ def main(argv=None):
     except Exception as e:
         if spack.config.get('config:debug'):
             raise
-        tty.die(e)
+        tty.error(e)
+        return 3
 
     except KeyboardInterrupt:
```

Comment:
I thought `KeyboardInterrupt` was supposed to be moved to the second `except`?

Reply:
I think the order we have now is safe because `KeyboardInterrupt` (and `SystemExit`) [inherit from BaseException rather than Exception](https://docs.python.org/3/library/exceptions.html). But I agree that changing the order from more-to-less specific would read better & potentially be more future proof, so I'll make that change now.